### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by bryanwweber**
Fix link to NumPy dev guidelines and Python 2/3 support

Changes proposed in this pull request:
- Update the CONTRIBUTING.md file

Just looking for some feedback on the changes to the Python bit at the end

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -85,7 +85,6 @@
 ## Python
 
 * Style generally follows PEP8 (https://www.python.org/dev/peps/pep-0008/)
-* Code in .py files needs to be written to work with both Python 2
-  and Python 3. Code in Cython files (.pyx or .pxd) should automatically work with both.
-* Code in the Python examples should be written for Python 3. Python 2 versions
-  are automatically generated as part of the build process
+* Code in .py and .pyx files needs to be written to work Python 3
```
Comment (speth) [inline CONTRIBUTING.md:88]: missing 'with'

Comment (speth) [reply CONTRIBUTING.md:88]: Also, probably want to use backticks around the file extensions, even if we didn't before.

**Review by speth — COMMENTED** [review]
Looks fine to me, with a couple if minor corrections. We might also want to say that Python code can be written to make use of features introduced in Python 3.4 or earlier.

**Comment by bryanwweber** [comment]
Merged in e0d35095970dbfef53b17c8bf3a8b376d2c49af6

**Pull request closed without merging**